src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/ServiceUtil.java

```diff
@@ -0,0 +1,75 @@
+package uk.gov.ons.ctp.integration.rhsvc.service.impl;
+
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.util.UUID;
+import uk.gov.ons.ctp.common.domain.CaseType;
+import uk.gov.ons.ctp.common.event.EventPublisher;
+import uk.gov.ons.ctp.common.event.EventPublisher.Channel;
+import uk.gov.ons.ctp.common.event.EventPublisher.EventType;
+import uk.gov.ons.ctp.common.event.EventPublisher.Source;
+import uk.gov.ons.ctp.common.event.model.Address;
+import uk.gov.ons.ctp.common.event.model.CollectionCase;
+import uk.gov.ons.ctp.common.event.model.CollectionCaseNewAddress;
+import uk.gov.ons.ctp.common.event.model.NewAddress;
+import uk.gov.ons.ctp.common.time.DateTimeUtil;
+import uk.gov.ons.ctp.integration.rhsvc.representation.CaseRequestDTO;
+
+/**
+ * This class aims to prevent code duplication by holding code which is common to more than one
+ * service.
+ */
+public class ServiceUtil {
+  private static final Logger log = LoggerFactory.getLogger(ServiceUtil.class);
+
+  static CollectionCase createCase(
+      CaseRequestDTO request, CaseType caseType, String collectionExerciseId) {
+    CollectionCase newCase = new CollectionCase();
+
+    newCase.setId(UUID.randomUUID().toString());
+    newCase.setCollectionExerciseId(collectionExerciseId);
+    newCase.setHandDelivery(false);
+    newCase.setSurvey("CENSUS");
+    newCase.setCaseType(caseType.name());
+    newCase.setAddressInvalid(false);
+    newCase.setCeExpectedCapacity(caseType == CaseType.CE ? 1 : 0);
```

Comment:
Don't know if this is right or not. Not clear why you would set this to 1 when caseType CE.  Obviously it has to be >= 1 , but is this just satisfying the latest checks, without knowing the real value ?   Perhaps a comment might help.

Reply:
Comment added:
// If a CE then it must have at least 1 resident